int64 len
CorridorLocalMsg[] corridors